---
title: "Snp_GWAS"
author: "Saira Asif"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# PACKAGES
library(tidyverse)

```

**Formating to vcf formate**

The text file provided by Neogen contains the SNPs for each sample (pig). Metadata is provided which includes HW p-value, Miner Allele Frequency (MAF), etc. The two files must be combined and formated to a vcf file for further analysis.


```{r}
### SNPS - E.coli Obervation Trail
# Reading in the snps txt file
snps <- read_tsv('SNP_FinalReport.txt')
# renaming the columns
names(snps) <- c('SNP_name', "Sample_ID", "Allele1_Forward", 'Allele2_Forward', 'Allele1_Top', 'Allele2_Top', 'Allele1_AB', 'Allele2 - AB', 'Contrast', 'Size', 'Confidence', 'SNP_Classification')

# Reading in the metadata file
meta1 <- read_tsv('SNP_Summary.txt')

# the values in this column are not unique
length(unique(snps$SNP_name)) == length(snps$SNP_name)

# all the values are unique 
length(unique(meta1$probeset_id)) == length(meta1$probeset_id)

# Filter metadata 
meta2 <- subset(meta1, probeset_id %in% unique(snps$SNP_name))

# Merge the two dataframe
# Combined dataframe
snps_meta <- inner_join(snps, meta2, by=c('SNP_name'='probeset_id'))

# Filter the snps based on HW p-value and MAF

```

